subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -255,8 +255,10 @@ static void tls_debug(void *ctx, int level, const char *file,
 		}
 	}
 
-	NET_DBG("%s:%04d: |%d| %s", basename, line, level,
-		log_strdup(str));
+	char *log_str = log_strdup(str);
+
+	/* Skip the extra EOL to improve log readability. */
+	NET_DBG("%s:%04d: |%d| %.*s", basename, line, level, strlen(log_str) - 1, log_str);
```

Comment:
Do you know that the `%.*s` is working on all the supported libc? I am asking because I was trying to use this in other part of the system but at that time (long time ago) it was not supported. So if this is really supported now also for minimal libc, then that is great.